cmd/operator/main.go

```diff
@@ -65,9 +65,10 @@ func main() {
 
 	for i := range deployments {
 		deployment := deployments[i]
-		apiKind := deployment.Labels["apiKind"]
-		if userconfig.KindFromString(apiKind) == userconfig.RealtimeAPIKind ||
-			userconfig.KindFromString(apiKind) == userconfig.AsyncAPIKind {
+		apiKind := userconfig.KindFromString(deployment.Labels["apiKind"])
+		if apiKind == userconfig.RealtimeAPIKind ||
+			(apiKind == userconfig.AsyncAPIKind && deployment.Labels["cortex.dev/async"] != "gateway") {
```

Comment:
Note to reviewer: I think this should have been checking `!= "gateway"` already (not related to this PR), but not 100% sure. Lmk what you think.

Reply:
I believe this is correct. The gateway is currently scaled by HPA. There is an if statement in UpdateMetricsCron and UpdateAutoscalerCron that ignores the gateway deployments. Perhaps `UpdateAutoscalerCron` and `UpdateMetricsCron` should be renamed to `UpdateAPIAutoscalerCron` and  `UpdateAPIMetricsCron` respectively to make it clear. If it were renamed, I would be comfortable with dropping the if statements in both functions